geti_sdk/rest_clients/dataset_client.py

```diff
@@ -157,3 +159,90 @@ def _media_folder_has_dataset_subfolders(self, path_to_folder: str) -> bool:
+        """
+        Return the media in the training dataset for the `model`, for
+        the specified `subset`. Subset can be `training`, `validation` or `testing`.
+
+        :param model: Model for which to get the media in the training dataset
+        :param subset: The subset for which to return the media items. Can be either
+            `training` (the default), `validation` or `testing`
+        return: A tuple containing:
+            - A list of all images in the specified subset of the training dataset of
+                the model
+            - A list of all video frames in the specified subset of the training
+                dataset of the model
+        """
+        ds_info = model.training_dataset_info
+        dataset_storage_id = ds_info.get("dataset_storage_id", None)
+        revision_id = ds_info.get("dataset_revision_id", None)
+        if dataset_storage_id is None or revision_id is None:
+            raise ValueError(
+                f"Unable to fetch the required dataset information from the model. "
+                f"Expected dataset and revision id's, got {ds_info} instead."
+            )
+        post_data = {
+            "condition": "and",
+            "rules": [{"field": "subset", "operator": "equal", "value": subset}],
+        }
+
+        images: MediaList[Image] = MediaList([])
+        video_frames: MediaList[VideoFrame] = MediaList([])
+
+        next_page = f"{self.base_url}/{dataset_storage_id}/training_revisions/{revision_id}/media:query"
+        while next_page:
+            response = self.session.get_rest_response(
+                url=next_page, method="POST", data=post_data
+            )
+            next_page = response.get("next_page", None)
+            for item in response["media"]:
+                if item["type"] == "image":
+                    item.pop("annotation_scene_id", None)
+                    item.pop("editor_name", None)
+                    item.pop("roi_id", None)
+                    image = MediaRESTConverter.from_dict(item, Image)
+                    images.append(image)
+
+                if item["type"] == "video":
+                    video_id = item["id"]
+                    frame_page = f"{self.base_url}/{dataset_storage_id}/training_revisions/{revision_id}/media/videos/{video_id}:query"
+                    while frame_page:
+                        frames_response = self.session.get_rest_response(
+                            url=frame_page, method="POST", data=post_data
+                        )
+                        for frame in frames_response["media"]:
+                            frame["video_id"] = video_id
+                            video_frame = MediaRESTConverter.from_dict(
+                                frame, VideoFrame
+                            )
+                            video_frames.append(video_frame)
+                        frame_page = frames_response.get("next_page", None)
```

Comment:
should it be `next_page`?

Reply:
`next_page` is already in use, changed it to `next_frame_page`